sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -412,26 +434,27 @@ AZ_NODISCARD az_result setup_response_redirect(CURL * p_curl, az_span * response
 
  * @return AZ_OK if request was sent and a response was received
  */
-AZ_NODISCARD az_result az_http_client_send_request_impl_process(
+static AZ_NODISCARD az_result _az_http_client_curl_send_request_impl_process(
     CURL * p_curl,
     az_http_request * p_request,
     az_http_response * response) {
   az_result result = AZ_ERROR_ARG;
 
-  AZ_RETURN_IF_CURL_FAILED(setup_headers(p_curl, p_request));
+  AZ_RETURN_IF_FAILED(_az_http_client_curl_setup_headers(p_curl, p_request));
 
-  AZ_RETURN_IF_CURL_FAILED(setup_url(p_curl, p_request));
+  AZ_RETURN_IF_FAILED(_az_http_client_curl_setup_url(p_curl, p_request));
 
-  AZ_RETURN_IF_CURL_FAILED(setup_response_redirect(p_curl, &response->_internal.http_response));
+  AZ_RETURN_IF_FAILED(
+      _az_http_client_curl_setup_response_redirect(p_curl, &response->_internal.http_response));
 
   if (az_span_is_equal(p_request->_internal.method, az_http_method_get())) {
-    result = az_curl_send_get_request(p_curl);
+    result = _az_http_client_curl_send_get_request(p_curl);
   } else if (az_span_is_equal(p_request->_internal.method, az_http_method_post())) {
-    result = az_curl_send_post_request(p_curl, p_request);
+    result = _az_http_client_curl_send_post_request(p_curl, p_request);
   } else if (az_span_is_equal(p_request->_internal.method, az_http_method_delete())) {
-    result = az_curl_send_delete_request(p_curl, p_request);
+    result = _az_http_client_curl_send_delete_request(p_curl, p_request);
   } else if (az_span_is_equal(p_request->_internal.method, az_http_method_put())) {
-    result = az_curl_send_upload_request(p_curl, p_request);
+    result = _az_http_client_curl_send_upload_request(p_curl, p_request);
   } else {
     return AZ_ERROR_HTTP_INVALID_METHOD_VERB;
   }
```

Comment:
below: Don't access internal fields directly and I'm questioning AZ_SPAN_ZEROBYTE